app/controllers/activities_controller.rb

```diff
@@ -30,4 +29,38 @@ def poll
       format.js
     end
   end
+
+  private
+
+  def period_start
+    DateTime.parse params[:period_start]
+  end
+
+  def period_end
+    if params[:period_end].empty?
+      DateTime.now
+    else
+      DateTime.parse params[:period_end]
+    end
+  end
+
+  def filtering_params
+    params.slice(:user, :type, :period_start, :period_end)
+  end
+
+  def filter_activities(activities)
+    filtering_params.each do |k, value|
```

Comment:
why `k` and `value` rather than `k` and `v` or `key` and `value`?

Reply:
i changed it to `key` and `value`

Reply:
>i changed it to key and value

I don't see this change.

Reply:
Must have slipped with all of the changes. Now it's there.